### PR TITLE
Fixes for teuthology-node-cleanup

### PR DESCRIPTION
**Pull request by zmc**
This mainly gets the command working without having to invoke with `--owner` for each user who has jobs running.

### CODE REVIEW

File: scripts/node_cleanup.py [hunk 1]
```diff
@@ -13,13 +13,21 @@ def main():
     stale = query.find_stale_locks(args.owner)
     if not stale:
         return
+    by_owner = {}
+    for node in stale:
+        if args.owner and node['locked_by'] != args.owner:
+            # This should not happen
```
Comment (kamoltat) [inline scripts/node_cleanup.py:19]: Should we log this like Error: "node shouldn't be locked by other people"

**Comment by kamoltat** [comment]
Some context as to why you are fixing this in the PR description would be super helpful, thanks!

File: scripts/node_cleanup.py (hunk 1 above)
Comment (zmc) [reply scripts/node_cleanup.py]: Added

File: scripts/node_cleanup.py [hunk 2]
```diff
@@ -13,13 +13,24 @@ def main():
...
+    by_owner = {}
+    for node in stale:
+        if args.owner and node['locked_by'] != args.owner:
+            log.warning(
+                f"Node {node['name']} expected to be locked by {args.owner} "
+                f"but found {node['locked_by']} instead"
+            )
+            continue
+        owner_nodes = by_owner.setdefault(node['locked_by'], [])
+        owner_nodes.append(node)
```
Comment (kamoltat) [inline scripts/node_cleanup.py:25]: I know that owner_nodes is a reference to by_owner, can't we just purely use by_owner? Am I missing something?

**Review by kamoltat — APPROVED** [review]
Other than the minor comment, LGTM

File: scripts/node_cleanup.py (hunk 2 above)
Comment (zmc) [reply scripts/node_cleanup.py:25]: `by_owner` is a dict where keys are e.g. "scheduled_zack@teuthology", and values are lists of dicts containing node data, whereas `owner_nodes` is a reference to one of those lists - but you're right that there's no real point in holding this reference.

I'll replace the above two lines with:
```
by_owner.setdefault(node['locked_by'], []).append(node)
```

**Review by kamoltat — APPROVED** [review]
LGTM, thanks for addressing the comment

**Pull request merged**